classes/publication/Repository.php

```diff
@@ -253,11 +252,11 @@ public function edit(Publication $publication, array $params)
                     }
 
                 // Otherwise generate a new thumbnail if a cover image exists
-                } elseif (!empty($newCoverImage)) {
-                    $coverImageFilePath = $publicFileManager->getContextFilesPath($submission->getData('contextId')) . '/' . $newCoverImage['uploadName'];
+                } elseif (!empty($newCoverImage) && $newCoverImage['temporaryFileId']) {
```

Comment:
just one more tiny comment: to check if 'temporaryFileId' key exists in the array $newCoverImage -- else there will be a PHP warning, I think... ?

Reply:
Sure, fixed by using `array_key_exists`